aggregation/quantile/tdigest/tdigest.go

```diff
@@ -297,7 +297,9 @@ func (d *tDigest) upperBound(index int) float64 {
 func (d *tDigest) appendCentroid(centroids []Centroid, c Centroid) []Centroid {
 	if len(centroids) == cap(centroids) {
 		newCentroids := d.centroidsPool.Get(2 * len(centroids))
-		newCentroids = append(newCentroids, centroids...)
+		for i := 0; i < len(centroids); i++ {
```

Comment:
The newly allocated slice is a zero-length slice with bigger capacity, don't think you can use `copy` here.